crates/core/tests/client.rs

```diff
@@ -128,6 +128,52 @@ mod tests {
         assert_eq!(num, 1);
     }
 
+    #[tokio::test]
+    async fn test_get_caller_address() {
```

Comment:
currently calling the testing solidity with the testsequencer. ideal flow would be to have an input solidity devs are comfortable with that is ran against our ci.